chaco/tools/range_selection.py

```diff
@@ -679,11 +678,10 @@ def _determine_axis(self):
                 return 0
 
     @observe([
-        trait("component").then(trait("index_mapper") | trait("value_mapper")),
-        trait("_plot").then(
-            trait("index_mapper", optional=True) \
-            | trait("value_mapper", optional=True)
-        ),
+        "component.index_mapper",
+        "component.value_mapper",
+        "_plot.index_mapper",
+        "_plot.value_mapper"
```

Comment:
I actually don't think we want this `optional=True` here.  `_plot = Trait(None, Any)`.
Without the `optional=True` everything seems to work fine, and this might be confusing.  whatever `_plot` gets set to should definitely have `index_mapper` and `value_mapper` traits. 

It is possible I am missing something here and we need to specify them as optional for some reason, but the `_plot` trait is guaranteed to exist.  If it is `None` it will not have `index_mapper` and `value_mapper` traits, but if it is ever set it should.  And it appears that there are not problems encountered when it is `None` (ie observe isn't complaining raising an error or warning or anything)

Reply:
I am happy to undo this revert if needed.  @rahulporuri lmk what you think on this

Reply:
can't we just use `plot.index_mapper` and `plot.value_mapper`?

Reply:
I tried doing this but the added test fails
But I can look into it more to try to better understand why thats happening